articles/npm-registry-opensource-code-security-with-npm6/index.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/npm-registry-opensource-code-security-with-npm6/
+title: Open-source packages & Code Security using npm
+description: The article addresses how opening DevTools, setting breakpoints, and restarting app instances is helpful for debugging. Since Node.js apps can fail without error messages.
+author: joseph-chege
+date: 2020-09-10T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/npm-registry-opensource-code-security-with-npm6/hero.jpg
+    alt: Devtools example image Node.js debugger
+---
+Security is a huge concern for everyone. The safety of the code you use is important regardless of you being a beginner or expert in Node.js app development. It is even a professional habit that developers acquire as they grow with experience. Making sure open-source codes are secure is one of the fundamental mission of NPM, with approaches to security at scale to empower NPM users to develop secure code.
+<!--more-->
+
+With NPM(short for Node Package Manager), you do not need to worry about the safety of your code. NPM provides vulnerability-scanning tools that are built-in your Node.js workflow. These tools are faster and they automatically review every install request you make, it checks the Node Security Platform vulnerably database, and warns you if you try to use unsafe codes. It even analyzes your dependencies trees to check specifically what has security issues and recommend a replacement or fix the issues automatically.
```

Comment:
@kimkimani 
This sentence is a bit difficult to read - not sure what it was meant to say - please revisit and rephrase. 
"it checks the Node Security Platform vulnerably database"